### ISSUE TITLE
Extend Webhook autoscaler to send fleet metadata with the request

### ISSUE THREAD

**Issue by swermin** [googleforgames/agones#3951]
**Is your feature request related to a problem? Please describe.**
We are running our fleets throughout different clusters and independently of one and other and use the webhook autoscaler to make sure that we have necessary sizes on all the fleets.
With this solution we are allocating game servers based on labels and not on fleet names, this means that on the `webhook autoscaler` endpoint we have no idea to determine the size since the request does not contain any metadata at all regarding the fleet in question.

**Describe the solution you'd like**
We want to extend the `FleetAutoscaleRequest` to include the fleet metadata when reporting to the service(s).

**Describe alternatives you've considered**
We made changes to the URL used for the `webhook` but that means that we have to update the URL every time we add new label and/or annotations

**Comment by markmandel** [comment]
Seems legit to me! I assume, name, annotations and labels?

**Comment by swermin** [comment]
That is correct. I was looking into the code and found that the easiest way to add these was to put the whole metadata into the request, but I do like the idea of making that payload smaller

**Comment by markmandel** [comment]
🤔 it would make the code easier to grab the entire metadata object and chuck it across. 

If you're implementing it, I'll let you decide 😄

**Comment by markmandel** [comment]
PR seems to be in flight. @swermin are you still working on this?

**Comment by swermin** [comment]
Hi! Yes I am, or I was. Sorry it took some time for this whole thing. Last few weeks have been busy so I didn’t have time to get to implementing e2e tests. My plan was to try wrap it up these coming days.

I have a general question though, do I need to make a new PR to get the example autoscaler to support this flow? Or is it about to get the fixes in and some one to help me get a new container image built to test with?

**Comment by markmandel** [comment]
> I have a general question though, do I need to make a new PR to get the example autoscaler to support this flow? 

Nope, you can add them to the exiting PR.

> Or is it about to get the fixes in and some one to help me get a new container image built to test with?

ooh, good question. Yeah, likely with help. Have a think about what is needed for the e2e tests, and if you need a new container or not for it.

**Comment by markmandel** [comment]
Given how tricky it is to do your own e2e testing image - @igooch WDYT of letting #3957 be merged (although it needs some feature gates), and then can be manually tested and/or we leave this ticket open to do a follow up with an e2e test?
